### PR TITLE
Added mlflow docker for tracking models

### PR DESCRIPTION
**Pull request by Hsankesara**

### CODE REVIEW

File: distributed-dcompose-stack/mlflow/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    mlflow-server:
+        restart: always
+        build:
+          context: ../../images/mlflow-server/
+          args:
+            - DB_URI=postgresql+psycopg2://${POSTGRES_USER}:${POSTGRES_PASSWORD}@pgdb:5432/mlflow-db
+            - SERVER_HOST=${HOST}
+            - SERVER_PORT=${MLFLOW_PORT}
+            - AWS_BUCKET_NAME=${AWS_BUCKET_NAME}
```
Comment (yatharthranjan) [inline distributed-dcompose-stack/mlflow/docker-compose.yml:34]: why not specify the args of mlflow directly here itself and in the Dockerfile you can have
```
ENTRYPOINT ["mlflow", "server"]
```

Then you don't need the Start.sh script and we can add any other mlflow arguments if we need later without any modifications.

File: distributed-dcompose-stack/mlflow/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ports:
+          - 9000:9000
+        environment:
+          - MINIO_ACCESS_KEY=${AWS_ACCESS_KEY_ID}
+          - MINIO_SECRET_KEY=${AWS_SECRET_ACCESS_KEY}
+        command: server /date
+        networks:
+          - backend
+        volumes:
+          - ${MP_MLFLOW_DIR}/s3:/date
```
Comment (yatharthranjan) [inline distributed-dcompose-stack/mlflow/docker-compose.yml:63]: it that a typo on `date`?

File: distributed-dcompose-stack/mlflow/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            - SERVER_PORT=${MLFLOW_PORT}
+            - AWS_BUCKET_NAME=${AWS_BUCKET_NAME}
+        environment:
+            - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+            - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+            - AWS_DEFAULT_REGION=${AWS_REGION}
+            - MLFLOW_S3_ENDPOINT_URL=http://s3:9000
+        ports:
+            - ${MLFLOW_PORT}:${MLFLOW_PORT}
+        expose:
```
Comment (yatharthranjan) [inline distributed-dcompose-stack/mlflow/docker-compose.yml:42]: You don't need this as long as it is exposed in the Dockerfile

File: distributed-dcompose-stack/mlflow/etc/s3/create_bucket.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+minioUrlHostWithPort = minioUrl.split('//')[1]
+print('[*] minio url: ',minioUrlHostWithPort)
+
+s3Client = Minio(
+    minioUrlHostWithPort,
+    access_key=accessID,
+    secret_key=accessSecret,
+    secure=False
+)
```
Comment (yatharthranjan) [inline distributed-dcompose-stack/mlflow/etc/s3/create_bucket.py:36]: I think you can add region here too to make it compatible with AWS S3 too (if someone wants to use managed services on AWS)

**Review by yatharthranjan — APPROVED** [review]
Good work. please find some comments below.

**Comment by Hsankesara** [comment]
Thanks, will implement those changes.

File: distributed-dcompose-stack/mlflow/docker-compose.yml (hunk 2 above)
Comment (Hsankesara) [reply distributed-dcompose-stack/mlflow/docker-compose.yml:63]: Yeah, it should be `data`

**Pull request merged**